### ISSUE TITLE
add support for django 3.2

### ISSUE THREAD

**Issue by pdpinch** [mitodl/edx-sga#303]
edX has published a guide for upgrading repos to django 3.2: https://openedx.atlassian.net/wiki/spaces/AC/pages/1704591765/Django+3.2+Upgrade+Steps+to+Upgrade+a+Repository

### acceptance criteria

- [ ] read the [steps to upgrade a repository](https://openedx.atlassian.net/wiki/spaces/AC/pages/1704591765/Django+3.2+Upgrade+Steps+to+Upgrade+a+Repository) and update this checklist with the steps that need to be done for this repo.

**Comment by natabene** [comment]
Please update [https://github.com/edx/upgrades/issues/71](https://github.com/edx/upgrades/issues/71) with a comment once this is done.

**Comment by umarmughal824** [comment]
@pdpinch could you please confirm which specific python version are we going to target here `3.8` or `3.9`?

**Comment by pdpinch** [comment]
We have to target the version that edx-platform uses, which I'm pretty sure is `3.8`.

**Comment by umarmughal824** [comment]
Yeah edx-platform still using python 3.8 https://github.com/mitodl/edx-platform/blob/master/Dockerfile#L62

**Comment by umarmughal824** [comment]
@pdpinch I have added the checklist in the PR description but there is not much difference between the edX suggested list. Just a difference is that we are already using Github actions rather than Travis.

**Comment by pdpinch** [comment]
Thanks for adding the checklist @umarmughal824. We should get started on this. Do you think we can do this as one issue, or should we create sub-issues?

**Comment by umarmughal824** [comment]
> should

@pdpinch I will check it tomorrow. If it will not be a big PR then we can go with one PR or vice versa.

**Comment by umarmughal824** [comment]
@natabene @pdpinch I saw that edX is still using `django=2.2` even on `master` that's the blocker.

**Comment by pdpinch** [comment]
How exactly is that blocking us?

I don’t think you’re going to see Django 3.2 in master until all the dependencies (like edx-sga) are upgraded. Maybe there’s a branch that we can use for testing purposes?

**Comment by umarmughal824** [comment]
> How exactly is that blocking us? I don’t think you’re going to see Django 3.2 in master until all the dependencies (like edx-sga) are upgraded. Maybe there’s a branch that we can use for testing purposes?

That's good to know. In that case, I will manually install django 3.2, then will test my PR over it.

**Comment by pdpinch** [comment]
There is a "chicken or the egg" problem here, because no one can run edx-platform with django 3.2 until all the libraries are updated. But we can't update this library without testing with an edx-platform on django 3.2. 

Can you leave a comment on edx/upgrades#71  explaining which commit on this repo we believe will work with django 3.2. That way the edX team can let us know when they have a django 3.2 candidate that we can test with.
